src/implementation.ts

```diff
@@ -80,120 +80,8 @@ function callResource(resourceConfig: ResourceConfig, resourcePath: ReadonlyArra
-             * loaders.foo.load({ foo_id: 4, include_private_data: false });
-             * \`\`\`
-             *
-             * If we collected everything up and tried to send the one
-             * request to the resource as a batch request, how do we know
-             * what the value for "include_private_data" should be? We're
-             * going to have to group these up up and send two requests to
-             * the resource to make sure we're requesting the right stuff.
-             *
-             * e.g. We'd need to make the following set of underlying resource
-             * calls:
-             *
-             * \`\`\`js
-             * foo({ foo_ids: [ 2 ], include_private_data: true });
-             * foo({ foo_ids: [ 3, 4 ], include_private_data: false });
-             * \`\`\`
-             *
-             * ~~~ tl;dr ~~~
-             *
-             * When we have calls to .load with differing non batch key args,
-             * we'll need to send multiple requests to the underlying
-             * resource to make sure we get the right results back.
-             *
-             * Let's create the request groups, where each element in the
-             * group refers to a position in "keys" (i.e. a call to .load)
-             *
-             * Example:
-             *
-             * \`\`\`js
-             * partitionItems('bar_id', [
-             *   { bar_id: 7, include_extra_info: true },
-             *   { bar_id: 8, include_extra_info: false },
-             *   { bar_id: 9, include_extra_info: true },
-             * ])
-             * \`\`\`
-             *
-             * Returns:
-             * \`[ [ 0, 2 ], [ 1 ] ]\`
-             *
-             * We could also have more than one batch key.
-             *
-             * Example:
-             *
-             * \`\`\`js
-             * partitionItems(['bar_id', 'properties'], [
-             *   { bar_id: 7, properties: ['property_1'], include_extra_info: true },
-             *   { bar_id: 8, properties: ['property_2'], include_extra_info: false },
-             *   { bar_id: 9, properties: ['property_3'], include_extra_info: true },
-             * ])
-             * \`\`\`
-             *
-             * Returns:
-             * \`[ [ 0, 2 ], [ 1 ] ]\`
-             *
-             * We'll refer to each element in the group as a "request ID".
-             */
-            let requestGroups;
-
-            if (${typeof resourceConfig.propertyBatchKey === 'string'}) {
-                requestGroups = partitionItems([
-                    '${resourceConfig.newKey}',
-                    '${resourceConfig.propertyBatchKey}'
-                ], keys);
-            } else {
-                requestGroups = partitionItems('${resourceConfig.newKey}', keys);
-            }
-
+function getBatchResourceCommonLoader(resourceConfig: BatchResourceConfig, resourcePath: ReadonlyArray<string>) {
```

Comment:
As I understand it, there's now 3 top level loader implementation functions:
- getPropertyBatchLoader
- getBatchLoader
- getNonBatchLoader

which makes sense.

The naming of "getBatchResourceCommonLoader" makes me think it might be another top level factory function (it follows the same `getXYZLoader` pattern), but I guess it's not - it's a helper shared between `getPropertyBatchLoader` and `getBatchLoader`? So maybe we could just tweak the name to clarify this?

maybe something like:

```suggestion
function batchLoaderLogic(resourceConfig: BatchResourceConfig, resourcePath: ReadonlyArray<string>) {
```

Reply:
Thanks!